web/auth/oauth.go

```diff
@@ -238,9 +239,57 @@ func (a *AuthorizeHTTPHandler) authorizeForm(c echo.Context) error {
 		return c.Redirect(http.StatusFound, u.String()+"#")
 	}
 
+	if !params.client.Flagship {
+		flags, err := feature.GetFlags(inst)
+		if err != nil {
+			return err
+		}
+
+		if clientsLimit, ok := flags.M["cozy.oauthclients.max"].(float64); ok && clientsLimit >= 0 {
+			limit := int(clientsLimit)
+
+			clients, _, err := oauth.GetConnectedUserClients(inst, 100, "")
+			if err != nil {
+				return fmt.Errorf("Could not get user OAuth clients: %w", err)
+			}
+			count := len(clients)
+
+			if count >= limit {
+				var manageDevicesURL, premiumURL string
+
+				connectedDevicesURL := inst.SubDomain(consts.SettingsSlug)
+				connectedDevicesURL.Fragment = "/connectedDevices"
+				manageDevicesURL = connectedDevicesURL.String()
+
+				if enablePremiumLinks, ok := flags.M["enable_premium_links"].(bool); ok && enablePremiumLinks {
+					var err error
```

Comment:
[nit] I think you can omit this line, as `err` is already declared before

Reply:
I remember getting a linter error without it but I don't anymore.